catroid/src/main/java/org/catrobat/catroid/content/actions/CloneAction.kt

```diff
@@ -20,28 +20,20 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-package org.catrobat.catroid.content.actions;
-
-import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
-
-import org.catrobat.catroid.content.Sprite;
-import org.catrobat.catroid.stage.StageActivity;
-
-public class CloneAction extends TemporalAction {
-
-	private Sprite sprite;
-
-	@Override
-	protected void update(float percent) {
-		if (sprite == null) {
-			return;
-		}
-
-		StageActivity.stageListener.cloneSpriteAndAddToStage(sprite);
-	}
-
-	public void setSprite(Sprite sprite) {
-		this.sprite = sprite;
-	}
-}
+package org.catrobat.catroid.content.actions
+
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.stage.StageActivity
+
+class CloneAction : TemporalAction() {
+    lateinit var sprite: Sprite
```

Comment:
It would be better if you initialize it in this situation with: var sprite: Sprite?, as the parameter is set from a java class which could possibly be null.